lib/vector-core/buffers/src/disk_v2/reader.rs

```diff
@@ -457,12 +488,26 @@ where
         // we'll simply wait for the writer to signal to us that progress has been made, which
         // implies a data file existing.
         loop {
+            let (reader_file_id, writer_file_id) = self.ledger.get_current_reader_writer_file_id();
             let data_file_path = self.ledger.get_current_reader_data_file_path();
             let data_file = match File::open(&data_file_path).await {
                 Ok(data_file) => data_file,
                 Err(e) => match e.kind() {
                     ErrorKind::NotFound => {
-                        self.ledger.wait_for_writer().await;
+                        // SCREAM SCREAM SCREAM
+                        // this is a not entirely proven out thought:
+                        // if the writer/reader file IDs are not the same, and we try to open our
+                        // current file and it doesn't exist, what actually happened is that we
+                        // deleted it as instructed but never made it to the "now update our reader
+                        // file ID in the ledger" step.  we should just increment our read file ID
+                        // and try again.
```

Comment:
I buy this, but it's one of those things I'd _really_ like some property testing to help verify.

Reply:
Yeah, agreed.

I'm gonna update the doc comment to more succinctly explain the invariant, but also add a comment about explicitly adding a test to prove this invariant for us.